enumerables.rb

```diff
@@ -0,0 +1,141 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+    if block_given?
+      item = 0
+      while item < size
+        yield(self[item])
+        item += 1
+      end
+    end
+
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+    if block_given?
+      index = 0
+      while index < size
+        yield(self[index], index)
+        index += 1
+      end
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+    if block_given?
+      filtered = []
+      my_each { |x| filtered.push(x) if yield(x) }
+    end
+    filtered
+  end
+
+  def my_all?(pattern = nil)
+    not_all = 0
+    if pattern.class == Regexp
+      my_each { |x| not_all = 1 if pattern.match(x).nil? }
+      not_all != 1
+    end
+    if block_given?
+      index = 0
+      while index < size
+        not_all = 1 unless yield(self[index])
+        index += 1
+        break if not_all == 1
+      end
+
+    end
+    not_all != 1
+  end
+
+  def my_any?(pattern = nil, &block)
+    any = 0
+    if pattern.class == Regexp
+      my_each { |x| any = 1 unless pattern.match(x).nil? }
+      any == 1
+    end
+    if block_given?
+      any = 1 unless my_none?(pattern = nil, &block)
+      any == 1
+    end
+  end
+
+  def my_none?(pattern = nil)
+    some = 0
+    if pattern.class == Regexp
+      my_each { |x| some = 1 unless pattern.match(x).nil? }
+      some == 1
+    end
```

Comment:
Please add the following functionality:

* when a pattern other than Regex or a Class is given => returns true only if none of the collection matches the pattern

These two should behave the same:
```
irb(main):037:0> %w[dog door rod blade].none?('dog')
=> false
irb(main):038:0> %w[dog door rod blade].my_none?('dog')
=> true
```

[`REQUIRED`]